rust/lance-encoding-datafusion/src/zone.rs

```diff
@@ -0,0 +1,211 @@
+        let zone_maps_buffer = encoded_zone_maps.try_to_mini_lance()?;
+        Ok(vec![EncodedMetadataBuffer {
+            parts: vec![zone_maps_buffer],
+        }])
+    }
+}
+
+impl FieldEncoder for ZoneMapsFieldEncoder {
+    fn maybe_encode(
+        &mut self,
+        array: ArrayRef,
+    ) -> Result<Vec<lance_encoding::encoder::EncodeTask>> {
+        // TODO: If we do the zone map calculation as part of the encoding task then we can
+        // parallelize statistics gathering.  Could be faster too since the encoding task is
+        // going to need to access the same data (although the input to an encoding task is
+        // probably too big for the CPU cache anyways).  We can worry about this if we need
+        // to improve write speed.
+        self.update(&array)?;
+        self.items_encoder.maybe_encode(array)
+    }
+
+    fn flush(&mut self) -> Result<Vec<lance_encoding::encoder::EncodeTask>> {
+        if self.cur_offset > 0 {
+            // Create final map
+            self.new_map()?;
+        }
+        self.items_encoder.flush()
+    }
+
+    fn finish(&mut self) -> BoxFuture<'_, Result<Vec<EncodedMetadataBuffer>>> {
+        async move { self.maps_to_metadata().await }.boxed()
+    }
+
+    fn num_columns(&self) -> u32 {
+        self.items_encoder.num_columns()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use arrow_array::types::Int32Type;
+    use arrow_schema::DataType;
+    use lance_datagen::{BatchCount, RowCount};
+    use lance_encoding::encoder::{CoreFieldEncodingStrategy, FieldEncoder, FieldEncodingStrategy};
+
+    #[tokio::test]
+    async fn test_basic_stats() {
+        let encoding_strategy = CoreFieldEncodingStrategy::default();
+        let inner = encoding_strategy
+            .create_field_encoder(
+                &encoding_strategy,
+                &DataType::Int32,
+                0,
+                4096,
+                true,
+                &HashMap::new(),
+            )
+            .unwrap();
+        let mut encoder =
+            super::ZoneMapsFieldEncoder::try_new(inner, DataType::Int32, 100).unwrap();
+
+        let gen = lance_datagen::gen()
+            .anon_col(lance_datagen::array::step::<Int32Type>())
+            .into_reader_rows(RowCount::from(1024), BatchCount::from(7));
+
+        for batch in gen {
+            let batch = batch.unwrap();
+            let array = batch.column(0);
+            encoder.maybe_encode(array.clone()).unwrap();
+        }
+
+        let zone_maps_buffer = encoder.finish().await.unwrap();
+        assert_eq!(zone_maps_buffer.len(), 1);
+        let zone_maps_buffer = zone_maps_buffer.into_iter().next().unwrap();
+        assert_eq!(zone_maps_buffer.parts.len(), 1);
+        let zone_maps_buffer = zone_maps_buffer.parts.into_iter().next().unwrap();
+        // TODO: Once reading is available we can check the contents of the zone maps buffer
+        // TODO: Test out the different types
```

Comment:
One type that always has special considerations are floats. Will want to document how we want to handle +/-0, Infinity, NaN. For V1, we wrote down the semantics here: https://lancedb.github.io/lance/format.html#statistic-values